core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsCredentialsProviderChain.java

```diff
@@ -199,13 +237,25 @@ public void setCredentialsProviders(Collection<? extends AwsCredentialsProvider>
         }
 
         @Override
-        public Builder credentialsProviders(AwsCredentialsProvider... credentialsProviders) {
-            return credentialsProviders(Arrays.asList(credentialsProviders));
+        public Builder credentialsIdentityProviders(
+                Collection<? extends IdentityProvider<? extends AwsCredentialsIdentity>> credentialsProviders) {
+            this.credentialsProviders = new ArrayList<>(credentialsProviders);
+            return this;
+        }
+
+        public void setCredentialsIdentityProviders(
+                Collection<? extends IdentityProvider<? extends AwsCredentialsIdentity>> credentialsProviders) {
+            credentialsIdentityProviders(credentialsProviders);
+        }
+
+        @Override
+        public Builder credentialsProviders(IdentityProvider<? extends AwsCredentialsIdentity>... credentialsProviders) {
```

Comment:
credentialsIdentityProviders ? This API input is `IdentityProvider<? extends AwsCredentialsIdentity>... credentialsProviders`
Sorry it was not clear to me how we have named the API as credentialsProviders or credentialsIdentityProviders .

Reply:
We are overloading the same method name where possible. This is what we are doing in all other places where new Identity type is accepted (see https://github.com/aws/aws-sdk-java-v2/pull/3829). Here, only in the Collection input case where overloading was not possible, we used `credentialsIdentityProvider()`. Using separate names in all places, could lead to confusion that they are 2 separate _properties_ of the chain. This is what we decided on in surface area review.